client/src/Classifier.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef classifier_h
+#define classifier_h
+
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <opencv2/ml/ml.hpp>
+
+using namespace std;
+using namespace cv;
+
+class Classifier {
+  CvSVM svm;
+
+public:
+  Classifier(string filename) {
+    svm.load(filename.c_str());
+  }
+
+  // Guess the characters name from the image icon.
+  string classify(const Mat &image) {
+    auto class_id = (size_t) svm.predict(Classifier::convert_image(image));
```

Comment:
Since this is basically just a "let's learn a few things about C++" PR, I'd also say that you should generally prefer the explicit casts (e.g. `static_cast<size_t>`) instead of the C-style casts, since the compiler can do better compile-time checks and stuff.